tidy3d/components/data/sim_data.py

```diff
@@ -1115,11 +1115,21 @@ def process_adjoint_sources(self, adj_srcs: list[SourceType]) -> AdjointSourceIn
             hashes_to_src_times[tmp_src_hash].append(src.source_time)
 
         num_ports = len(hashes_to_src_times)
-        num_unique_freqs = len({src.source_time.freq0 for src in adj_srcs})
+        unique_freqs = {src.source_time.freq0 for src in adj_srcs}
+        num_unique_freqs = len(unique_freqs)
 
         # next, figure out which treatment / normalization to apply
         if num_unique_freqs == 1:
             log.info("Adjoint source creation: one unique frequency, no normalization.")
+            freqs_adj = self.simulation.freqs_adjoint
+
+            # if many adjoint freqs, but only 1 unique, need to mask out the non-contributors
+            if len(freqs_adj) > 1:
+                coords = dict(f=freqs_adj)
+                data = [1 if f == tuple(unique_freqs)[0] else 0 for f in freqs_adj]
+                post_norm = xr.DataArray(data, coords=coords)
+                return AdjointSourceInfo(sources=adj_srcs, post_norm=post_norm, normalize_sim=True)
```

Comment:
So `post_norm` is either `0` or `1`, and it's only `1` if the frequency is the central frequency of the adjoint source? Is the adjoint `source_time.freq0` always equal to the single frequency that was selected out?

Reply:
yes to all.